File: backend/app/schemas/training.py
"""Training schemas for request/response validation."""

from typing import List, Optional
from pydantic import BaseModel, Field


class TrainingRequest(BaseModel):
    """Schema for training request."""
    prompt_content: str = Field(..., min_length=1, description="用户输入的原始 Prompt")
    model: str = Field(default="deepseek", description="目标模型")
    scenario: str = Field(default="general", description="使用场景")


class Mistake(BaseModel):
    """A single mistake identified in the user's prompt."""
    mistake: str = ""
    why_it_matters: str = ""
    how_to_fix: str = ""


class NextPractice(BaseModel):
    """A practice task for the user."""
    model_config = {"protected_namespaces": ()}
    task: str = ""
    instruction: str = ""
    model_suggestion: str = ""


class TrainingFeedback(BaseModel):
    """Training feedback with model-aware coaching."""
    user_level: str = "beginner"
    main_mistakes: List[Mistake] = []
    learning_formula: str = ""
    next_practice: NextPractice = Field(default_factory=NextPractice)
    short_feedback: str = ""


class RadarScore(BaseModel):
    """Radar chart dimension score — 8 dimensions including model-aware metrics."""
    model_config = {"protected_namespaces": ()}
    clarity: float = 0.0
    specificity: float = 0.0
    creativity: float = 0.0
    feasibility: float = 0.0
    completeness: float = 0.0
    conciseness: float = 0.0
    model_fit: float = 0.0
    token_efficiency: float = 0.0


class TokenDiagnosis(BaseModel):
    """Token redundancy diagnosis."""
    total_tokens: int = 0
    redundant_tokens: int = 0
    redundant_ratio: float = 0.0
    redundant_details: List[str] = []
    suggestions: List[str] = []


class OptimizedVersion(BaseModel):
    """An optimized version of the prompt."""
    title: str = ""
    content: str = ""
    focus: str = ""


class TrainingAdvice(BaseModel):
    """Training coach advice."""
    model_config = {"protected_namespaces": ()}
    summary: str = ""
    strengths: List[str] = []
    weaknesses: List[str] = []
    improvements: List[str] = []
    model_specific_tips: str = ""
    scenario_tips: str = ""


class TrainingTask(BaseModel):
    """A training task for the user."""
    title: str = ""
    description: str = ""
    difficulty: str = "beginner"
    hint: str = ""
    expected_improvement: str = ""


class TrainingResponse(BaseModel):
    """Complete training response."""
    radar_scores: RadarScore = Field(default_factory=RadarScore)
    token_diagnosis: TokenDiagnosis = Field(default_factory=TokenDiagnosis)
    optimized_versions: List[OptimizedVersion] = []
    training_advice: TrainingAdvice = Field(default_factory=TrainingAdvice)
    training_tasks: List[TrainingTask] = []
    training_feedback: TrainingFeedback = Field(default_factory=TrainingFeedback)
    original_prompt: str = ""
    selected_model: str = ""
    scenario: str = ""

File: frontend/src/api/training.ts
import apiClient from './client'

export interface RadarScores {
  clarity: number
  specificity: number
  creativity: number
  feasibility: number
  completeness: number
  conciseness: number
  model_fit: number
  token_efficiency: number
}

export interface TokenDiagnosis {
  total_tokens: number
  redundant_tokens: number
  redundant_ratio: number
  redundant_details: string[]
  suggestions: string[]
}

export interface OptimizedVersion {
  title: string
  content: string
  focus: string
}

export interface TrainingAdvice {
  summary: string
  strengths: string[]
  weaknesses: string[]
  improvements: string[]
  model_specific_tips: string
  scenario_tips: string
}

export interface TrainingTask {
  title: string
  description: string
  difficulty: string
  hint: string
  expected_improvement: string
}

export interface Mistake {
  mistake: string
  why_it_matters: string
  how_to_fix: string
}

export interface NextPractice {
  task: string
  instruction: string
  model_suggestion: string
}

export interface TrainingFeedback {
  user_level: string
  main_mistakes: Mistake[]
  learning_formula: string
  next_practice: NextPractice
  short_feedback: string
}

export interface TrainingResponse {
  radar_scores: RadarScores
  token_diagnosis: TokenDiagnosis
  optimized_versions: OptimizedVersion[]
  training_advice: TrainingAdvice
  training_tasks: TrainingTask[]
  training_feedback: TrainingFeedback
  original_prompt: string
  selected_model: string
  scenario: string
}

export interface TrainingRequest {
  prompt_content: string
  model: string
  scenario: string
}

export interface ModelOption {
  value: string
  label: string
  description: string
}

export interface ScenarioOption {
  value: string
  label: string
  description: string
}

export const trainingAPI = {
  analyze: async (data: TrainingRequest): Promise<TrainingResponse> => {
    const response = await apiClient.post<TrainingResponse>('/training/analyze', data)
    return response.data
  },

  getModels: async (): Promise<ModelOption[]> => {
    const response = await apiClient.get<{ models: ModelOption[] }>('/training/models')
    return response.data.models
  },

  getScenarios: async (): Promise<ScenarioOption[]> => {
    const response = await apiClient.get<{ scenarios: ScenarioOption[] }>('/training/scenarios')
    return response.data.scenarios
  },
}
